### ISSUE TITLE
Inconsistent summaries for `run --dry`in provision, prepare

### ISSUE THREAD

**Issue by lukaszachy** [psss/tmt#552]
One test, one plan. One runs `tmt run --dry` to see if all is more-less OK before real execution.

Since nothing should be done (--dry) all summaries should contain 0.
```
$ tmt run --dry
/var/tmp/tmt/run-058
 
/some-plan
    discover
        how: fmf
        directory: XXXX
        filter: component:ZZZZ
        summary: 0 tests selected
    provision
        how: virtual
        image: fedora
        user: root
        memory: 2048 MB
        disk: 10 GB
        summary: 1 guest provisioned
    prepare
        how: shell
        name: enable module
        overview: 1 script found
        how: install
        name: install PKG
        summary: 2 preparations applied
    execute
        how: tmt
        summary: 0 tests executed
    report
        how: display
        summary: no results found
    finish
        summary: 0 tasks completed
```

**Comment by FrNecas** [comment]
I agree that `--dry` needs to be improved but shouldn't it be the other way around? When all summaries are going to show 0, aren't you missing information to check if the run is OK so seeing the number of preparations that would be done if it wasn't dry mode, could actually be useful?

I've tried using dry run on a very simple example and got this:

```
/var/tmp/tmt/run-021

/plan
    discover
        how: fmf
        directory: /home/fnecas/devel/tmttest
        summary: 0 tests selected
    provision
        how: local
        summary: 1 guest provisioned
    prepare
        summary: 1 preparations applied
    execute
        how: tmt
        summary: 0 tests executed
    report
        how: display
        summary: no results found
    finish
        summary: 0 tasks completed
```

I feel like like seeing  the number of guests and preparations is useful. However, I'd also expect to see the number of discovered tests and perhaps even in execute seeing the number of tests that would be executed (so 1 in my simple case instead of 0). What do you think?

**Comment by psss** [comment]
> I'd also expect to see the number of discovered tests

Yeah, and here the question is where we put the line between the dry mode and real action. For example running the discover step for the integration plan in the `fmf` repository using `tmt run -dddnvvv discover plan -n integration` results in the following:

```
discover
    workdir: /var/tmp/tmt/run-352/plans/integration/discover
    how: fmf
    order: 50
    url: https://github.com/psss/tmt
        Clone 'https://github.com/psss/tmt' to '/var/tmp/tmt/run-352/plans/integration/discover/default/tests'.
        Run command 'git clone https://github.com/psss/tmt /var/tmp/tmt/run-352/plans/integration/discover/default/tests'.
    ref: fedora
        Checkout ref 'fedora'.
        Run command 'git checkout -f fedora'.
        Run command 'git rev-parse --short HEAD'.
    filter: tier: 1, 2
        Check metadata tree in '/var/tmp/tmt/run-352/plans/integration/discover/default/tests/'.
    summary: 0 tests selected
    status: done
    Write file '/var/tmp/tmt/run-352/plans/integration/discover/step.yaml'.
    Write file '/var/tmp/tmt/run-352/plans/integration/discover/tests.yaml'.
```

Which nicely shows what would happen (we should still improve wording of the `run()` method to use `Would run command` to make this completely clear) but under the hood actually nothing happens which also means we cannot list discovered tests...

**Comment by lukaszachy** [comment]
We have to fix the inconsistency. Can we agree on user story behind --dry and fix the behaviour accordingly? Especially if/how much 'actions' we allow to happen (Download remote repo in discover? Get real image name for fedora in provision, ...)

I'd love to see what would happen but I'm not sure were to mark the line.

**Comment by psss** [comment]
Yeah, seems we need to agree on a clear line for the `--dry` option scope. Seems like a minor problem though, adding as a `could` have issue and marking for discussion.

**Comment by psss** [comment]
Brainstorm what could be the meaning:

* Show what would happen but do not do any changes (just reading the state of the world)
* Don't touch network at all (no importing remote plans)
* No guest provision (no local vm or container creation)
* Do not submit to the server (e.g. schedule a beaker job)
* No permanent changes on the test runner 

### Specific Cases

* discover fmf & git cloning ... no, git clone, just show 
* tmt plan show & remote plans ... the `--shallow` option could be rename to something like `--no-network-access`

On possibility would be to introduce separate options for identified "areas" like `--no-network-access` or `--no-task-creation`.

**Comment by therazix** [comment]
Closing as duplicate of #4443
